docs/development/releasing.rst

```diff
@@ -274,30 +277,43 @@ The procedure for this is straightforward:
 
       $ git branch v<version>.x
 
-#. Render the changelog with towncrier, as above.
-
-#. Tag this commit using the next major version followed by ``.dev``. For example,
-   if you have just branched ``4.0``, create the ``v4.1.dev`` tag on the commit
-   adding the ``4.1`` section to the changelog::
-
-      $ git tag -s "v<next_version>.dev" -m "Back to development: v<next_version>"
-
-#. Also update the "what's new" section of the docs to include a section for the
+#. Update the "what's new" section of the docs to include a section for the
    next major version.  E.g.::
 
       $ cp docs/whatsnew/<current_version>.rst docs/whatsnew/<next_version>.rst
 
    You'll then need to edit ``docs/whatsnew/<next_version>.rst``, removing all
-   the content but leaving the basic structure.  You may also need  to
+   the content but leaving the basic structure.  You may also need to
    replace the "by the numbers" numbers with "xxx" as a reminder to update them
    before the next release. Then add the new version to the top of
    ``docs/whatsnew/index.rst``, update the reference in ``docs/index.rst`` to
-   point to the that version, and commit these changes ::
+   point to the that version.
+
+#. Update the "what's new" section of the current version,
+  ``docs/whatsnew/<current_version>.rst``, and remove all content, replacing it
+  with::
+
+      :orphan:
 
+      `What's New in Astropy <current_version>?
+      <https://docs.astropy.org/en/v<current_version>/whatsnew/<current_version>.html>`__
```

Comment:
One small issue with this is that it results in a temporarily broken link until v5.0 is actually released.